package-res/ccc/core/base/axis/color-axis.js

```diff
@@ -60,11 +71,81 @@ def('pvc.visual.ColorAxis', pvc_Axis.extend({
             return this.base(scale);
         },
 
+        // CDF603 
+        // set PreserveMap default to true
+        // must be called before the first read of PreserveMap? 
+        setPreserveColorMap: function() {
+            this.option.defaults({'PreserveMap': true});
+        },
+
+
+        // CDF603
+        // returns the stored Map if it is supposed to be 
+        // preserved and if it exists
+        _haveMap: function() {
+              if(this.option('PreserveMap') && this.Map) 
+                return this.Map;  
+        },
+
+ 
+        //CDF603
+        // given a colorMap it substitutes it if it exists
+        // a previously stored one
+        effectiveMap: function(colorMap , baseScheme) {
+            var prevMap = this._haveMap(),
+                newMap;
+
+            if(prevMap) newMap = prevMap;
+            else newMap = colorMap ;
+
+            return newMap;
+        },
+
+        //CDF603
+        // given a scheme, computes the corresponding colorMap
+        // scheme is a function that for a given element of the 
+        // domain returns the corresponding color object (pv.Color)
+        _getMapFromScheme: function(scheme) { 
+            
+            var domain = this.domainValues();
+
+                var newMap = this.Map || {};
+                domain.forEach(
+                    function(d) { 
+                        if(!newMap[d]) newMap[d] = scheme(d); 
```

Comment:
Testing property existence like this is vulnerable to errors due to the domain values being the name of standard `Object.prototype` properties. To guard against this problem, please test this way:
```javascript
if(!def.hasOwn(newMap, key)) newMap[key] = scale(key);
```